core/trino-main/src/main/java/io/trino/likematcher/LikeMatcher.java

```diff
@@ -25,54 +25,37 @@
 
 public class LikeMatcher
 {
-    private final String pattern;
-    private final Optional<Character> escape;
-
     private final int minSize;
     private final OptionalInt maxSize;
     private final byte[] prefix;
     private final byte[] suffix;
     private final Optional<Matcher> matcher;
 
     private LikeMatcher(
-            String pattern,
-            Optional<Character> escape,
             int minSize,
             OptionalInt maxSize,
             byte[] prefix,
             byte[] suffix,
             Optional<Matcher> matcher)
     {
-        this.pattern = pattern;
-        this.escape = escape;
         this.minSize = minSize;
         this.maxSize = maxSize;
         this.prefix = prefix;
         this.suffix = suffix;
         this.matcher = matcher;
     }
 
-    public String getPattern()
-    {
-        return pattern;
-    }
-
-    public Optional<Character> getEscape()
-    {
-        return escape;
-    }
-
-    public static LikeMatcher compile(String pattern)
+    public static LikePatternMatcher compile(String pattern)
     {
         return compile(pattern, Optional.empty(), true);
     }
 
-    public static LikeMatcher compile(String pattern, Optional<Character> escape)
+    public static LikePatternMatcher compile(String pattern, Optional<Character> escape)
```

Comment:
This should still return a `LikeMatcher` -- i.e., LikeMatcher should not have to understand LikePattern (the wrapper) at all. That's a planner/interpreter-level concern. The caller should wrap it on the outside.